recipes/jsoncpp/all/test_package/CMakeLists.txt

```
@@ -2,10 +2,14 @@ cmake_minimum_required(VERSION 3.1)
 project(test_package)
 
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup()
+conan_basic_setup(TARGETS)
 
 find_package(jsoncpp REQUIRED CONFIG)
 
 add_executable(${PROJECT_NAME} test_package.cpp)
-target_link_libraries(${PROJECT_NAME} jsoncpp::jsoncpp_lib) # TODO: remove jsoncpp:: namespace when fixed in conanfile.py of jsoncpp
+if(JSONCPP_SHARED)
+    target_link_libraries(${PROJECT_NAME} jsoncpp_lib)
+else()
+    target_link_libraries(${PROJECT_NAME} jsoncpp_static)
+endif()
```

This approach does not require a cmake definition:
```suggestion
if(TARGET jsoncpp_lib)
    target_link_libraries(${PROJECT_NAME} jsoncpp_lib)
else()
    target_link_libraries(${PROJECT_NAME} jsoncpp_static)
endif()
```

---

yes, this is what I would eventually do as a consumer, but it doesn't test jsoncpp_static target since jsoncpp_lib target is always defined.

---

You'd need to flip the logic and check `if target shared`